qa/smoke-test-http/src/test/java/org/elasticsearch/http/SystemIndexRestIT.java

```diff
@@ -0,0 +1,172 @@
+            searchRequest.addParameter("allow_system_index_access", "true");
+
+            // Disallow no indices to cause an exception if the flag above doesn't work
+            searchRequest.addParameter("allow_no_indices", "false");
+            searchRequest.setJsonEntity("{\"query\": {\"match\":  {\"some_field\":  \"some_value\"}}}");
+
+            final Response searchResponse = getRestClient().performRequest(searchRequest);
+            assertThat(searchResponse.getStatusLine().getStatusCode(), is(200));
+            Map<String, Object> responseMap = entityAsMap(searchResponse);
+            assertThat(responseMap, hasKey("count"));
+            assertThat(responseMap.get("count"), equalTo(1));
+        });
+
+        // now try without `allow_system_index_access`
+        assertDeprecationWarningOnAccess(SystemIndexTestPlugin.SYSTEM_INDEX_NAME, SystemIndexTestPlugin.SYSTEM_INDEX_NAME);
+
+        // And with a partial wildcard
+        assertDeprecationWarningOnAccess(".test-*", SystemIndexTestPlugin.SYSTEM_INDEX_NAME);
+
+        // And with a total wildcard
+        assertDeprecationWarningOnAccess(randomFrom("*", "_all"), SystemIndexTestPlugin.SYSTEM_INDEX_NAME);
+
+        // Try to index a doc directly
+        {
+            String expectedWarning = "this request accesses system indices: [" + SystemIndexTestPlugin.SYSTEM_INDEX_NAME + "], but in a " +
+                "future major version, direct access to system indices will be prevented by default";
+            RequestOptions expectWarningOptions = RequestOptions.DEFAULT.toBuilder().setWarningsHandler(expectedWarning::equals).build();
+            Request putDocDirectlyRequest = new Request("PUT", "/" + SystemIndexTestPlugin.SYSTEM_INDEX_NAME + "/_doc/43");
+            putDocDirectlyRequest.setJsonEntity("{\"some_field\":  \"some_other_value\"}");
+            putDocDirectlyRequest.setOptions(expectWarningOptions);
+            Response response = getRestClient().performRequest(putDocDirectlyRequest);
+            assertThat(response.getStatusLine().getStatusCode(), equalTo(201));
+        }
+    }
+
+    private void assertDeprecationWarningOnAccess(String queryPattern, String warningIndexName) throws IOException {
+        String expectedWarning = "this request accesses system indices: [" + warningIndexName + "], but in a " +
+            "future major version, direct access to system indices will be prevented by default";
+        RequestOptions expectWarningOptions = RequestOptions.DEFAULT.toBuilder()
+            .setWarningsHandler(w -> w.contains(expectedWarning) == false || w.size() != 1)
+            .build();
+        Request searchRequest = new Request("GET", "/" + queryPattern + randomFrom("/_count", "/_search"));
+        searchRequest.setJsonEntity("{\"query\": {\"match\":  {\"some_field\":  \"some_value\"}}}");
+        // Disallow no indices to cause an exception if this resolves to zero indices (as we expect)
```

Comment:
I think this comment is incorrect?

Reply:
Yes, this is from an old version of the test, before I switched this to emit a deprecation warning instead of omitting system indices.